pkg/scheduler/framework/runtime/framework.go

```diff
@@ -463,23 +498,50 @@ func (f *frameworkImpl) expandMultiPointPlugins(profile *config.KubeSchedulerPro
 			// the user intent is to override the default plugin or make some other explicit setting.
 			// Either way, discard the MultiPoint value for this plugin.
 			// This maintains expected behavior for overriding default plugins (see https://github.com/kubernetes/kubernetes/pull/99582)
-			if enabledSet.Has(ep.Name) {
+			if enabledSet.has(ep.Name) {
+				overridePlugins.insert(ep.Name)
 				klog.InfoS("MultiPoint plugin is explicitly re-configured; overriding", "plugin", ep.Name)
 				continue
 			}
 
 			// if this plugin is already registered via MultiPoint, then this is
 			// a double registration and an error in the config.
-			if multiPointEnabled.Has(ep.Name) {
+			if multiPointEnabled.has(ep.Name) {
 				return fmt.Errorf("plugin %q already registered as %q", ep.Name, pluginType.Name())
 			}
 
 			// we only need to update the multipoint set, since we already have the specific extension set from above
-			multiPointEnabled.Insert(ep.Name)
+			multiPointEnabled.insert(ep.Name)
 
 			newPlugins := reflect.Append(plugins, reflect.ValueOf(pg))
 			plugins.Set(newPlugins)
 		}
+
+		// Reorder plugins if needed. Here is the expected order:
+		// - part 1: overridePlugins. Their order stay intact as how they're specified in regular extension point.
+		// - part 2: multiPointEnabled - i.e., plugin defined in multipoint but not overriding regular extension point.
+		// - part 3: other plugins (excluded by part 1 & 2) in regular extension point.
+		newPlugins := reflect.New(reflect.TypeOf(e.slicePtr).Elem()).Elem()
+		// part 1
+		for _, name := range enabledSet.list {
+			if overridePlugins.has(name) {
+				newPlugins = reflect.Append(newPlugins, reflect.ValueOf(pluginsMap[name]))
+				enabledSet.delete(name)
+			}
+		}
+		// enabledSet.isEmpty() implies part 3 is empty, so don't need to continue
```

Comment:
If I understood correctly, we can harmless remove https://github.com/kubernetes/kubernetes/blob/8ef5ccdad6bba8b664b51a8b29fd1a5ce396a59d/pkg/scheduler/framework/runtime/framework.go#L532-L535 and https://github.com/kubernetes/kubernetes/blob/897f2f78a5c7ff659dfce23f53dd36b9261a23c9/pkg/scheduler/framework/runtime/framework.go#L480-L481

After removal, program still behaves the same but more clearly to read. Separate the reading and writing process.